projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/order-history/b2b-order-history-orders-flow.e2e-spec.ts

```diff
@@ -81,4 +85,61 @@ describe('Order History with orders', () => {
     cy.get('.cx-order-history-po a').should('contain', poNumber);
     cy.get('.cx-order-history-cost-center a').should('contain', costCenter);
   });
+
+  describe('Order details - reorder', () => {
+    it('should display order details page with the reorder button', () => {
+      cy.get('.cx-order-history-value').first().click();
+      cy.get('cx-order-details-reorder div div button').should(
+        'contain',
+        'Reorder'
+      );
+    });
+
+    it('items in the cart should match previous order when proceeding with reorder', () => {
+      // Saving cartId before starting the reorder flow
+      b2bCheckout.addB2bProductToCart();
+      cart.goToCart();
+      cart.saveCartId();
+
+      goToOrderDetails();
+      saveOrderDetails();
+
+      // Check if the reorder button exists
+      cy.get('cx-order-details-reorder div div button').first().click();
+      cy.get('.cx-reorder-dialog-areyousure-section').should('exist');
+
+      // Click on continue and wait for the response from backend
+      const cartFromOrderAlias = interceptCartFromOrderEndpoint();
+      cy.get('.cx-reorder-dialog-footer div button.btn-primary')
+        .first()
+        .click();
+      waitForResponse(cartFromOrderAlias);
+
+      // Go to cart and verify if the cartId is different
+      cart.goToCart();
+      cart.verifyCartIdIsDifferent();
+
+      // Count <li> and see if they match with order details
+      cy.get('@totalOrderHistoryListItems').then(
+        (totalOrderHistoryListItems: any) => {
+          cy.get('cx-cart-item-list .cx-item-list-row').should(
+            'have.length',
+            totalOrderHistoryListItems
+          );
+        }
+      );
```

Comment:
Perhaps using this selector would mahe the test more resilient to UI evolution.
`'cx-order-details-reorder button'`